compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -861,7 +861,7 @@ pub fn provide_both(providers: &mut Providers) {
     providers.dllimport_foreign_items = |tcx, krate| {
         let module_map = tcx.foreign_modules(krate);
         let module_map =
-            module_map.iter().map(|lib| (lib.def_id, lib)).collect::<FxHashMap<_, _>>();
+            module_map.iter().map(|(def_id, lib)| (def_id, lib)).collect::<FxHashMap<_, _>>();
```

Comment:
We can just remove lines 863 and 864 now since `foreign_modules` now returns the `FxHashMap` this constructs.